megamek/src/megamek/common/weapons/artillery/ISArrowIV.java

```diff
@@ -55,4 +59,35 @@ public ISArrowIV() {
                 .setReintroductionFactions(F_CC)
                 .setStaticTechLevel(SimpleTechLevel.ADVANCED);
     }
+
+    @Override
+    public int[] getRanges(Mounted weapon) {
+        // modify the ranges for Arrow missile systems based on the ammo selected
+        int minRange = getMinimumRange();
+        int sRange = getShortRange();
+        int mRange = getMediumRange();
+        int lRange = getLongRange();
+        int eRange = getExtremeRange();
+        boolean hasLoadedAmmo = (weapon.getLinked() != null);
+        if (hasLoadedAmmo) {
+            AmmoType atype = (AmmoType) weapon.getLinked().getType();
+            if (atype.getMunitionType().contains(AmmoType.Munitions.M_ADA)) {
+                minRange = 0;
+                sRange = 17;
```

Comment:
I'm seeing a lot of these magic numbers, let's define them as constants. Also, this logic block is the same for IS and CLArrowIV - ideally they'd be in one place.

Reply:
I'd considered introducing an ArrowIV.java base class; guess I'll bite that bullet.

Reply:
Done.